### PR TITLE
security: add permissions block to workflows

### PR DESCRIPTION
**Pull request by reakaleek**
## Details

⚠️ This PR was created by an automated tool. Please review the changes carefully. ⚠️ 

We want to set the default permissions for workflows to read-only for contents. 
This is a security measure to prevent accidental changes to the repository.

This change adds a top-level permissions block to all workflows in the .github/workflows directory.
```yaml
permissions:
  contents: read
```

In some cases workflows might need more permissions than just `contents: read`.
Please checkout this branch and add the necessary permissions to the workflows.

If your workflow uses a Personal Access Token (PAT), we can still add the permissions block,
   but it will not have any effect.

Merging this PR as is might cause workflows that need more permissions to fail.

If there are any questions, please reach out to the @elastic/observablt-ci

### CODE REVIEW

File: .github/workflows/test-reporter.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ on:
       - test
     types: [completed]
 
+permissions:
+  contents: read
+  actions: read
```
Comment (reakaleek) [inline .github/workflows/test-reporter.yml:14]:
```suggestion
  actions: read
  checks: write
```

File: .github/workflows/test-packages.yml [hunk 2]
```diff
@@ -20,6 +20,9 @@ on:
         required: false
         type: string
 
+permissions:
+  contents: read
```
Comment (reakaleek) [inline .github/workflows/test-packages.yml:24]:
```suggestion
  contents: read
  actions: read
```

File: .github/workflows/labeler.yml [hunk 3]
```diff
@@ -5,6 +5,9 @@ on:
   pull_request_target:
     types: [opened]
 
+permissions:
+  contents: read
```
Comment (reakaleek) [inline .github/workflows/labeler.yml:9]:
```suggestion
  contents: read
  issues: write
  pull-requests: write
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -6,6 +6,9 @@ name: build
 on:
   workflow_call: ~
 
+permissions:
+  contents: read
```
Comment (reakaleek) [inline .github/workflows/build.yml:10]:
```suggestion
  contents: read
  actions: write
```

File: .github/workflows/build-packages.yml [hunk 5]
```diff
@@ -6,6 +6,9 @@ name: build-packages
 on:
   workflow_call: ~
 
+permissions:
+  contents: read
```
Comment (reakaleek) [inline .github/workflows/build-packages.yml:10]:
```suggestion
  contents: read
  actions: write
```

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -6,6 +6,10 @@ name: build
...
+  actions: write
```
Comment (v1v) [inline .github/workflows/build.yml:11]: what's the reason for write? Does the upload-artifact requires it?

File: .github/workflows/test-packages.yml [hunk 7]
```diff
@@ -20,6 +20,10 @@ on:
...
+  actions: write
```
Comment (v1v) [inline .github/workflows/test-packages.yml:25]: does the download/artefacts require it?

File: .github/workflows/build.yml (hunk 6 above)
Comment (reakaleek) [reply .github/workflows/build.yml:11]: Thank you for questioning this. Yes, I was assuming so. But let me verify this.

Comment (reakaleek) [reply .github/workflows/build.yml:11]: You are right, it's not needed at all.

I don't remember anymore why I had this assumption.

File: .github/workflows/build-packages.yml [hunk 8]
```diff
@@ -6,6 +6,10 @@ name: build-packages
...
+  actions: write
```
Comment (reakaleek) [inline .github/workflows/build-packages.yml:11]:
```suggestion
```

File: .github/workflows/build.yml (hunk 6 above)
Comment (reakaleek) [inline .github/workflows/build.yml:11]:
```suggestion
```

File: .github/workflows/test-packages.yml (hunk 7 above)
Comment (reakaleek) [inline .github/workflows/test-packages.yml:25]:
```suggestion
```

**Review by v1v — APPROVED** [review]

**Comment by v1v** [comment]
Blocked by https://github.com/elastic/apm-agent-php/pull/1144

**Pull request merged**